files/pull-sandbox-image.sh

```diff
@@ -29,7 +40,7 @@ function retry() {
   done
 }
 
-ecr_password=$(retry aws ecr get-login-password --region $region)
+ecr_password=$(retry aws ecr get-login-password --region "${region}")
```

Comment:
if we add carter's fix https://github.com/awslabs/amazon-eks-ami/pull/1646/files and don't break the whole workflow on this error, could we have a branch for empty/non-empty `ecr_password` that will make the image pull with/without creds?

something roughly like
```bash
if [[ -z ${ecr_password} ]]; then
  retry sudo crictl pull "${sandbox_image}"
else
  retry sudo crictl pull --creds "AWS:${ecr_password}" "${sandbox_image}"
fi
```

Reply:
agree